crengine/src/lvtinydom.cpp

```
@@ -10724,11 +10728,309 @@ bool ldomDocument::findText( lString32 pattern, bool caseInsensitive, bool rever
         CRLog::debug("No text found: Range is empty");
         return false;
     }
-    return range.findText( pattern, caseInsensitive, reverse, words, maxCount, maxHeight, maxHeightCheckStartY );
+    return range.findText( pattern, caseInsensitive, reverse, words, maxCount, maxHeight, maxHeightCheckStartY, false, patternIsRegex );
+}
+
+#if USE_SRELL_REGEX !=1
+int checkRegex(const lString32 & searchPattern)
+{
+    return 0; // no error
+}
+
+int getAndClearRegexSearchError()
+{
+    return 0; // no error
+}
```

I think I'd prefer to not have these dummy implementations, and have the prototypes in lvtinydom.h themselves wrapped in a `#if USE_SRELL_REGEX == 1`.
(And if we were to have these dummy implementations, they should probably return an error instead of pretending all is fine :)

---

If we wrap the functions in `lvtinydom.h` we have the problem, that when compiling `cre.cpp` the `-DUSE_SRELL_REGEX` is not defined.

---

So may be with this ?
```diff
--- a/Makefile
+++ b/Makefile
@@ -154,9 +154,9 @@ endif
 $(OUTPUT_DIR)/libs/libkoreader-cre.so: cre.cpp \
                        $(if $(USE_LUAJIT_LIB),$(LUAJIT_LIB),) \
                        $(CRENGINE_LIB)
        $(CXX) -I$(CRENGINE_SRC_DIR)/crengine/include/ $(DYNLIB_CXXFLAGS) \
-               -DLDOM_USE_OWN_MEM_MAN=$(if $(WIN32),0,1) \
+               -DLDOM_USE_OWN_MEM_MAN=$(if $(WIN32),0,1) -DUSE_SRELL_REGEX=1 \
                $(if $(WIN32),-DQT_GL=1) $(SYMVIS_FLAGS) -static-libstdc++ -o $@ $^
 ifdef DARWIN
        install_name_tool -change \
                `otool -L "$@" | grep "libluajit" | awk '{print $$1}'` \
```
It adds some noise here, but it feels better than having it that strange in crengine.

---

Shall I add a `-DUSE_SRELL_REGEX` to the base Makefile Line 159
```
$(OUTPUT_DIR)/libs/libkoreader-cre.so: cre.cpp \
                        $(if $(USE_LUAJIT_LIB),$(LUAJIT_LIB),) \
                        $(CRENGINE_LIB)
        $(CXX) -I$(CRENGINE_SRC_DIR)/crengine/include/ $(DYNLIB_CXXFLAGS) \
                -DLDOM_USE_OWN_MEM_MAN=$(if $(WIN32),0,1) \
                -DUSE_SRELL_REGEX \
                $(if $(WIN32),-DQT_GL=1) $(SYMVIS_FLAGS) -static-libstdc++ -o $@ $^
ifdef DARWIN
```

---

Yep (see my previous comment :). But with `=1` as you check for `-DUSE_SRELL_REGEX=1`